OpenRA.Mods.Common/Orders/RepairOrderGenerator.cs

```diff
@@ -12,86 +12,191 @@
 			if (repairable != null)
-				repairBuilding = repairable.FindRepairBuilding(underCursor);
+				repairBuilding = repairable.FindRepairBuilding(actor);
 			else
 			{
-				var repairableNear = underCursor.TraitOrDefault<RepairableNear>();
+				var repairableNear = actor.TraitOrDefault<RepairableNear>();
 				if (repairableNear != null)
 				{
 					orderId = "RepairNear";
-					repairBuilding = repairableNear.FindRepairBuilding(underCursor);
+					repairBuilding = repairableNear.FindRepairBuilding(actor);
 				}
 			}
 
 			if (repairBuilding == null)
-				yield break;
+				return null;
 
-			yield return new Order(orderId, underCursor, Target.FromActor(repairBuilding), mi.Modifiers.HasModifier(Modifiers.Shift))
+			return new Order(orderId, actor, Target.FromActor(repairBuilding), mi.Modifiers.HasModifier(Modifiers.Shift))
 			{
-				VisualFeedbackTarget = Target.FromActor(underCursor)
+				VisualFeedbackTarget = Target.FromActor(actor)
 			};
 		}
 
-		protected override void Tick(World world)
+		protected override IEnumerable<IRenderable> Render(WorldRenderer wr, World world) { yield break; }
+		protected override IEnumerable<IRenderable> RenderAboveShroud(WorldRenderer wr, World world) { yield break; }
+		protected override IEnumerable<IRenderable> RenderAnnotations(WorldRenderer wr, World world)
 		{
-			if (world.LocalPlayer != null &&
-				world.LocalPlayer.WinState != WinState.Undefined)
-				world.CancelInputMode();
+			if (isDragging && (mousePos - dragStartMousePos).Length > Game.Settings.Game.SelectionDeadzone)
+			{
+				var lastMousePos = wr.Viewport.ViewToWorldPx(Viewport.LastMousePos);
+				var diag1 = wr.ProjectedPosition(lastMousePos);
+				var diag2 = wr.ProjectedPosition(dragStartMousePos);
+				yield return new RectangleAnnotationRenderable(diag1, diag2, diag1, 1, Color.White);
+				var actors = SelectActorsInBoxWithDeadzone(world, dragStartMousePos, lastMousePos);
+
+				if (actors != null && actors.Any())
+					foreach (var actor in actors)
+					{
+						if (actor != null)
+						{
+							var decorationBounds = actor.TraitsImplementing<IDecorationBounds>().ToArray();
+							var bounds = decorationBounds.FirstNonEmptyBounds(actor, wr);
+							yield return new SelectionBarsAnnotationRenderable(actor, bounds, true, false);
+						}
+					}
+			}
+
+			yield break;
 		}
 
-		protected override IEnumerable<IRenderable> Render(WorldRenderer wr, World world) { yield break; }
-		protected override IEnumerable<IRenderable> RenderAboveShroud(WorldRenderer wr, World world) { yield break; }
-		protected override IEnumerable<IRenderable> RenderAnnotations(WorldRenderer wr, World world) { yield break; }
+		static IEnumerable<Actor> SelectActorsInBoxWithDeadzone(World world, int2 a, int2 b)
+		{
+			// For dragboxes that are too small, choose nothing
+			// because the WorldInteractionControllerWidget can show the imformation
+			if ((a - b).Length <= Game.Settings.Game.SelectionDeadzone)
+				return null;
```

Comment:
If you return an empty Enumerable instead of null, you can remove the null and any check at line 150.

```suggestion
				return Enumerable.Empty<Actor>();
```

Reply:
ok